### ISSUE TITLE
Incorrect rule for sql pool name validation.

### ISSUE THREAD

**Issue by tsukashusan** [hashicorp/terraform-provider-azurerm#11554]
### The regular expression section of the source code below should be changed to the correct specification.

According to the following Azure resource naming conventions, **UTF-8 character encoding should be acceptable**.

https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules#microsoftsql
azurerm/internal/services/synapse/validate/sql_pool_name.go

```
func SqlPoolName(i interface{}, k string) (warnings []string, errors []error) {
	v, ok := i.(string)
	if !ok {
		errors = append(errors, fmt.Errorf("expected type of %s to be string", k))
		return
	}

	// The name attribute rules are :
	// 1. can contain only letters, numbers and underscore.
	// 2. The value must be between 1 and 15 characters long

	_if !regexp.MustCompile(`^[a-zA-Z_\d]{1,15}$`).MatchString(v) {
		errors = append(errors, fmt.Errorf("%s can contain only letters, numbers or underscore, The value must be between 1 and 15 characters long", k))
		return
	}_

	return warnings, errors
}
```

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform v0.14.10
+ provider registry.terraform.io/hashicorp/azurerm v2.57.0
+ provider registry.terraform.io/hashicorp/random v3.1.0

Your version of Terraform is out of date! The latest version
is 0.15.1. You can update by downloading from https://www.terraform.io/downloads.html

### Affected Resource(s)

* `azurerm_synapse_sql_pool`

### Terraform Configuration Files

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
resource "azurerm_synapse_sql_pool" "example" {
  name                 = "販売管理"
  synapse_workspace_id = azurerm_synapse_workspace.example.id
  sku_name             = "DW100c"
  create_mode          = "Default"
  collation            = "Japanese_CS_AS_KS_WS"
}
```

### Panic Output

**Error: name can contain only letters, numbers or underscore, The value must be between 1 and 15 characters long**

  on synapse-db.tf line 72, in resource "azurerm_synapse_sql_pool" "example":
  72: resource "azurerm_synapse_sql_pool" "example" {
### Expected Behaviour
Japanese(UTF-8 character) sql pool name is created

### Actual Behaviour
The message "panic out put" is displayed.

### Steps to Reproduce

1. `terraform plan`

### References

* #0000

**Comment by owenfarrell** [comment]
@tsukashusan - The documentation you are referencing is specific to Microsoft SQL, which seems to play by a different set of rules from Synapse.

Valid Synapse SQL Pool names are documented [here](https://docs.microsoft.com/en-us/sql/relational-databases/databases/database-identifiers#rules-for-regular-identifiers). In summary, SQL pool names are limited to letters, numbers, and a few special characters.


Reference - [Quickstart: Create a SQL Pool (step 3)](https://docs.microsoft.com/en-us/azure/synapse-analytics/sql-data-warehouse/create-data-warehouse-portal#create-a-sql-pool)

**Comment by tsukashusan** [comment]
Hi @owenfarrell ,
Thanks for the documentation reference.
I checked the documentation you gave me and found that
"A letter as defined by the Unicode Standard 3.2. The Unicode definition of letters includes Latin characters from a through z, from A through Z, and **also letter characters from other languages.**" It says
Therefore, the characters of each country must be included in Unicode Standard 3.2.

I hope you will check the following URL.

https://docs.microsoft.com/en-us/sql/relational-databases/databases/database-identifiers?toc=/azure/synapse-analytics/sql-data-warehouse/toc.json&bc=/azure/synapse-analytics/sql-data-warehouse/breadcrumb/toc.json&view=azure-sqldw-latest&preserve-view=true#rules-for-regular-identifiers
